File: Java/p10443.java

import java.util.Scanner;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
/**
 *
 * @author DarKnight98
 */
public class p10443 {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int kasus = input.nextInt();
        for (int i = 0; i < kasus; i++) {
            int r = input.nextInt();
            int c = input.nextInt();
            int d = input.nextInt();
            char[][] data = new char[r][c];

            for (int j = 0; j < data.length; j++) {
                data[j] = input.next().toCharArray();
            }

            char[][] res = new char[r][c];
            for (int j = 0; j < data.length; j++) {
                for (int k = 0; k < data[j].length; k++) {
                    res[j][k] = data[j][k];
                }
            }
            
            for (int l = 0; l < d; l++) {
                for (int j = 0; j < data.length; j++) {
                    for (int k = 0; k < data[j].length; k++) {
                        if (j == data.length - 1 && k != data[j].length-1) {
                            if (data[j][k] != data[j][k+1]) {
                                if (cekMenang(data[j][k], data[j][k+1])) {
                                    res[j][k+1] = data[j][k];
                                } else {
                                    res[j][k] = data[j][k+1];
                                }
                            }
                        } else {
                            if (k != data[j].length-1) {
                                if (data[j][k] != data[j][k+1]) {
                                    if (cekMenang(data[j][k], data[j][k+1])) {
                                        res[j][k+1] = data[j][k];
                                    } else {
                                        res[j][k] = data[j][k+1];
                                    }
                                }
                                if (data[j][k] != data[j+1][k]) {
                                    if (cekMenang(data[j][k], data[j+1][k])) {
                                        res[j+1][k] = data[j][k];
                                    } else {
                                        res[j][k] = data[j+1][k];
                                    }
                                }
                            } else {
                                if (j == data.length - 1) break;
                                if (data[j][k] != data[j+1][k]) {
                                    if (cekMenang(data[j][k], data[j+1][k])) {
                                        res[j+1][k] = data[j][k];
                                    } else {
                                        res[j][k] = data[j+1][k];
                                    }
                                }
                            }
                        }
                    }
                }
                for (int j = 0; j < data.length; j++) {
                    for (int k = 0; k < data[j].length; k++) {
                        data[j][k] = res[j][k];
                    }
                }
            }
            
            for (int j = 0; j < data.length; j++) {
                for (int k = 0; k < data[j].length; k++) {
                    System.out.print(res[j][k]);
                }
                System.out.println();
            }
            
            if (! (i == kasus-1))System.out.println();
        }
    }

    static boolean cekMenang(char char1, char char2) {
        if (char1 == 'R') {
            if (char2 == 'R' || char2 == 'S') {
                return true;
            } else {
                return false;
            }
        } else if (char1 == 'S') {
            if (char2 == 'S' || char2 == 'P') {
                return true;
            } else {
                return false;
            }
        } else {
            if (char2 == 'P' || char2 == 'R') {
                return true;
            } else {
                return false;
            }
        }
    }
}

File: C,C++/p10443.c
#include <stdio.h>

int cekMenang(char, char);

int main () {
	int r, c, d, kasus;
	scanf("%d", &kasus);
	int i;
	for (i = 0; i < kasus; i++) {
		scanf("%d %d %d", &r, &c, &d);
		char data [r+1][c+1];
		int j, k;
		for (j = 0; j < r; j++) {
			scanf("%s", data[j]);	
		}
		
//		printf("%c\n", data[r-1][c-1]);
//		printf("\n");
		char res [r+1][c+1];
		
		for (j = 0; j < d; j++) {
			int l;
			for (k = 0; k < r-1; k++) {
				for(l = 0; l < c; l++) {
					if (l==0) res[k][l] = data[k][l];
					if (l != c-1) {
						if (cekMenang(data[k][l], data[k][l+1])) res[k][l+1] = data[k][l];
						else res[k][l] = data[k][l+1];
						
						if (cekMenang(data[k][l], data[k+1][l])) res[k+1][l] = data[k][l];
						else res[k][l] = data[k+1][l];
					} else {
						if (cekMenang(data[k][l], data[k+1][l])) res[k+1][l] = data[k][l];
						else res[k][l] = data[k+1][l];
					}
				}
			}
			
			for (l = 0; l < c; l++) {
				if (l == c-1) res[k][l] = data[k][l];
				else {
					if (cekMenang(data[k][l], data[k][l+1])) res[k][l+1] = data[k][l];
				else res[k][l] = data[k][l+1];
				}				
			}
			
			for (k = 0; k < r; k++) {
				for (l = 0; l < c; l++) {
					data[k][l] = res[k][l];
				}
			}
			
			
		}
		
		for (k = 0; k < c; k++) {
				printf("%s\n", res[k]);
		}
	}
}

int cekMenang(char char1, char char2) {
	if (char1 == 'R') {
		if (char2 == 'R' || char2 == 'S') return 1;
		else return 0;
	} else if (char1 == 'S') {
		if (char2 == 'S' || char2 == 'P') return 1;
		else return 0;
	} else {
		if (char2 == 'P' || char2 == 'R') return 1;
		else return 0;
	}
}
